src/Adapter/MySQL.php

```diff
@@ -321,22 +326,98 @@ private function computeOrderByField(array $filterToTableMapping)
+        ) {
+            return $orderField;
+        }
+
+        $this->addSelectField('out_of_stock');
+
+        // order by out-of-stock last
+        $computedQuantityField = $this->computeFieldName('quantity', $filterToTableMapping);
+        $byOutOfStockLast = 'IFNULL(' . $computedQuantityField . ',0) <= 0';
+
+        /**
+         * Default behaviour when out of stock
+         * 0 - when deny orders
+         * 1 - when allow orders
+         *
+         * @var int
+         */
+        $isAvailableWhenOutOfStock = (int) Product::isAvailableWhenOutOfStock(2);
+
+        // computing values for order by 'allow to order last'
+        $computedField = $this->computeFieldName('out_of_stock', $filterToTableMapping);
+        $computedValue = $isAvailableWhenOutOfStock ? 0 : 1;
+        $computedDirection = $isAvailableWhenOutOfStock ? 'ASC' : 'DESC';
+
+        // order by allow to order last
+        $byOOPS = str_replace(
+            [':field', ':value', ':direction'],
+            [$computedField, $computedValue, $computedDirection],
+            'FIELD(:field, :value) :direction'
+        );
+
+        $orderField = $byOutOfStockLast . ', '
+            . $byOOPS . ', '
+            . $orderField;
+
+        return $orderField;
+    }
+
+    /**
+     * Add alias to table field name
+     *
+     * @param string $fieldName
+     * @param array $filterToTableMapping
+     *
+     * @return string Table Field name with an alias
+     */
+    private function computeFieldName($fieldName, $filterToTableMapping)
+    {
+        if (array_key_exists($fieldName, $filterToTableMapping)
             && (
                 // If the requested order field is in the result, no need to change tableAlias
                 // unless a fieldName key exists
-                isset($filterToTableMapping[$orderField]['fieldName'])
+                isset($filterToTableMapping[$fieldName]['fieldName'])
                 || $this->getInitialPopulation() === null
-                || !$this->getInitialPopulation()->getSelectFields()->contains($orderField)
+                || !$this->getInitialPopulation()->getSelectFields()->contains($fieldName)
             )
         ) {
-            $joinMapping = $filterToTableMapping[$orderField];
-            $orderField = $joinMapping['tableAlias'] . '.' . (isset($joinMapping['fieldName']) ? $joinMapping['fieldName'] : $orderField);
+            $joinMapping = $filterToTableMapping[$fieldName];
+            $fieldName = $joinMapping['tableAlias'] . '.' . (isset($joinMapping['fieldName']) ? $joinMapping['fieldName'] : $fieldName);
+
+            if (isset($joinMapping['aggregateFunction']) && isset($joinMapping['aggregateFieldName'])) {
+                $fieldName = $joinMapping['aggregateFunction'] . '(' . $fieldName . ') as ' . $joinMapping['aggregateFieldName'];
+            }
         } else {
-            $orderField = 'p.' . $orderField;
+            if (!strpos($fieldName, '(')) {
```

Comment:
What are you testing here, the 0 position or false?

Reply:
false.
If FieldName has '(' string than do nothing there

Reply:
```suggestion
            if (strpos($fieldName, '(') !== false) {
```
Otherwise if the `(` is in the first position, the test will pass and break the sql request

Reply:
Thanks 😁